core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -130,44 +138,47 @@ private void checkDiskSpaceAndHandleExceptions(DockerClient client) {
      * @param client an active Docker client
      */
     private void checkDiskSpace(DockerClient client) {
+        DiskSpaceUsage df = runInsideDocker(client, cmd -> cmd.withCmd("df", "-P"), (dockerClient, id) -> {
+                String logResults = dockerClient.logContainerCmd(id)
+                    .withStdOut(true)
+                    .exec(new LogToStringContainerCallback())
+                    .toString();
+
+                return parseAvailableDiskSpace(logResults);
+        });
+
+        log.info("Disk utilization in Docker environment is {} ({} )",
+            df.usedPercent.map(x -> x + "%").orElse("unknown"),
+            df.availableMB.map(x -> x + " MB available").orElse("unknown available"));
+
+        if (df.availableMB.orElseThrow(NotAbleToGetDiskSpaceUsageException::new) < 2048) {
+            log.error("Docker environment has less than 2GB free - execution is unlikely to succeed so will be aborted.");
+            throw new NotEnoughDiskSpaceException("Not enough disk space in Docker environment");
+        }
+    }
 
-        List<Image> images = client.listImagesCmd().exec();
-        if (!images.stream().anyMatch(it -> it.getRepoTags() != null && asList(it.getRepoTags()).contains("alpine:3.2"))) {
-            PullImageResultCallback callback = client.pullImageCmd("alpine:3.2").exec(new PullImageResultCallback());
-            callback.awaitSuccess();
+    public <T> T runInsideDocker(Consumer<CreateContainerCmd> createContainerCmdConsumer, BiFunction<DockerClient, String, T> block) {
```

Comment:
wish I could make it module-level, but I can't because the consumer is in `dockerclient` subpackage